.github/workflows/update-version.yml

```diff
@@ -0,0 +1,36 @@
+# This workflow automatically updates VERSION file in repo if any change is pushed to main branch. It performs below steps:
+# 1. Checks out the repo with depth 0 for full history.
+# 2. Runs make command to update VERSION file with the latest tag
+# 3. Commit and push updated VERSION file
+
+name: Update VERSION File
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  update-version-file:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Update VERSION file
+      run: make update-version
+
+    - name: Commit and push changes
+      run: |
+        if [ -n "$(git status --porcelain VERSION)" ]; then
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
```

Comment:
@ie-pham do you think this will work with our latest security changes to GHA?

i'll admit i'm not sure where the creds are that allow this to work?

Reply:
In terms of the GHA, I think this looks fine as it doesn't require any vault secrets. However, I am unsure if it is possible to push directly to main without a pull request.

Reply:
New GHA is not needed after fixing the comment from @zalegrala 👍